prov/gni/include/gnix_datagram.h

```diff
@@ -67,22 +67,24 @@ extern "C" {
  * exchange is completed.
  */
 
-/*
- * gnix_dgram_hndl - handle to a datagram management
- * instance.
- *
- * nic: pointer to gnix_cm_nic associated with this dgram_hndl
- * bnd_dgram_free_list: head of a linked list of available bnd dgrams
- * bnd_dgram_acive_list: head of linked list of active bnd dgrams
- * wc_dgram_free_list: head of linked list of available
- *			wildcard datagrams
- * wc_dgram_active_list: head of linked list of active
- *			wildcard datagrams
- * dgram_base: base address for allocated vector of dgrams
- * n_dgrams: number of bound dgrams
- * n_wc_dgrams: number of wildcard dgrams
+/**
+ * Datagram allocator struct
+ *
+ * @var cm_nic               pointer to a previously allocated cm_nic with
+ *                           which this datagram is associated
+ * @var bnd_dgram_free_list  head of free list for bound datagrams
+ * @var bnd_dgram_active_list  head of active list for bound datagrams
+ * @var wc_dgram_free_list   head of free list of wildcard datagrams
+ * @var wc_dgram_active_list head of active list of wildcard datagrams
+ * @var dgram_base           starting address of memory block from
+ *                           which datagram structures are allocated
+ * @var progress_thread      pthread id of progress thread for this
+ *                           datagram allocator
+ * @var n_dgrams             number of bound datagrams managed by the
+ *                           datagram allocator
+ * @var n_wc_dgrams          number of wildcard datagrams managed by
+ *                           the datagram allocator
  */
```

Comment:
yes please do if you haven't already opened an issue.  for the datagram we need dlist.